test/functional/tests/conftest.py

```diff
@@ -157,28 +164,44 @@ def base_prepare(item):
         kill_all_io()
         DeviceMapper.remove_all()
 
-        if installer.check_if_installed():
+        uninstall_cas = list(TestRun.item.iter_markers(name="uninstall_cas"))
+        install_rpm = get_rpm_param(item)
+        opencas_installed = Installer.check_if_installed()
+
+        if opencas_installed:
             try:
                 from api.cas.init_config import InitConfig
                 InitConfig.create_default_init_config()
                 unmount_cas_devices()
                 casadm.stop_all_caches()
                 casadm.remove_all_detached_cores()
             except Exception:
-                pass  # TODO: Reboot DUT if test is executed remotely
+                if TestRun.executor.is_remote():
+                    TestRun.executor.reboot()
 
         for disk in TestRun.dut.disks:
             disk.umount_all_partitions()
             create_partition_table(disk, PartitionTable.gpt)
 
-        if get_force_param(item) and not TestRun.usr.already_updated:
-            installer.rsync_opencas_sources()
-            installer.reinstall_opencas()
-        elif not installer.check_if_installed():
-            installer.rsync_opencas_sources()
-            installer.set_up_opencas()
+        if uninstall_cas and opencas_installed:
+            RpmInstaller.rsync_opencas(install_rpm=install_rpm)
+            RpmInstaller.uninstall_opencas()
+        elif get_force_param(item) and not TestRun.usr.already_updated:
+            RpmInstaller.rsync_opencas(install_rpm=install_rpm)
+            RpmInstaller.reinstall_opencas(install_rpm=install_rpm)
+        elif not opencas_installed and not uninstall_cas:
+            RpmInstaller.rsync_opencas(install_rpm=install_rpm)
+            RpmInstaller.set_up_opencas(install_rpm=install_rpm)
```

Comment:
If test has `uninstall` mark, but `force-reinstall` param is given, CAS will be installed anyway. IMO it sounds line misbehavior

Reply:
good point